clients/python/dd_client.py

```diff
@@ -1,118 +1,56 @@
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
 
 """
 
-try:
-    import urllib.request as urllib2
-except ImportError:
-    import urllib2
-
-try:
-    import http.client as httplib
-except ImportError:
-    import httplib
-
-
-import os.path
-import json
-import uuid
-import datetime
-
-
-VERBOSE=False
-DD_TIMEOUT = 2000 # seconds, for long blocking training alls, as needed
-
-def LOG(msg):
-    """Output a log message."""
-    # XXX: may want to use python log manager classes instead of this stupid print
-    if VERBOSE:
-        msg = str(datetime.datetime.now()) + ' ' + msg
-        print (msg)
+import requests
 
-### Exception classes :
-
-class DDCommunicationError(Exception):
-    def __init__(self, url, http_method,  headers, body, response=None):
-        self.msg = """DeepDetect Communication Error"""
-        self.http_method = http_method
-        self.req_headers = headers
-        self.req_body = body
-        self.url = url
-        self.res_headers = None
-        if response is not None:
-            self.res_headers = response.get_info()
-            
-    def __str__(self):
-        msg = "%s %s\n"%(str(self.http_method),str(self.url))
-        for h,v in self.req_headers.iteritems():
-            msg += "%s:%s\n"%(h,v)
-        msg += "\n"
-        if self.req_body is not None:
-            msg += str(self.req_body)[:100]
-        msg += "\n"
-        msg += "--\n"
-        msg += str(self.res_headers)
-        msg += "\n"
-        return msg
-
-class DDDataError(Exception):
-    def __init__(self, url, http_method,  headers, body, data=None):
-        self.msg = "DeepDetect Data Error"
-        self.http_method = http_method
-        self.req_headers = headers
-        self.req_body = body
-        self.url = url
-        self.data = data
-
-    def __str__(self):
-        msg = "%s %s\n"%(str(self.http_method),str(self.url))
-        if self.data is not None:
-            msg += str(self.data)[:100]
-        msg += "\n"
-        for h,v in self.req_headers.iteritems():
-            msg += "%s:%s\n"%(h,v)
-        msg += "\n"
-        if self.req_body is not None:
-            msg += str(self.req_body)
-        msg += "\n"
-        msg += "--\n"
-        msg += str(self.data)
-        msg += "\n"
-        return msg
+DD_TIMEOUT = 2000  # seconds, for long blocking training calls, as needed
 
 API_METHODS_URL = {
-    "0.1" : {
-        "info":"/info",
-        "services":"/services",
-        "train":"/train",
-        "predict":"/predict"
+    "0.1": {
+        "info": "/info",
+        "services": "/services",
+        "train": "/train",
+        "predict": "/predict"
     }
 }
 
+
 class DD(object):
     """HTTP requests to the DeepDetect server
-
     """
 
     # return types
-    RETURN_PYTHON=0
-    RETURN_JSON=1
-    RETURN_NONE=2
+    RETURN_JSON = 0
+    RETURN_TEXT = 1
+    RETURN_NONE = 2
```

Comment:
We cannot inverse semantic from what exists. Please keep `RETURN_JSON` for a JSON string, and `RETURN_PYTHON` for a Python dict.